File: src/main/java/com/intel/distml/util/DoubleArray.java
package com.intel.distml.util;

import com.intel.distml.api.DMatrix;
import com.intel.distml.api.Session;
import com.intel.distml.util.KeyCollection;

import java.util.HashMap;

/**
 * Created by yunlong on 12/8/15.
 */

public class DoubleArray extends SparseArray<Long, Double> {

    public DoubleArray(long dim) {
        super(dim, DataDesc.KEY_TYPE_LONG, DataDesc.ELEMENT_TYPE_DOUBLE);
    }

}

File: src/main/scala/com/intel/distml/util/scala/DoubleArray.scala
package com.intel.distml.util.scala

import com.intel.distml.util.DataDesc

/**
 * Created by yunlong on 1/3/16.
 */
class DoubleArray (
dim : Long) extends SparseArray[Long, Double](dim, DataDesc.KEY_TYPE_LONG, DataDesc.ELEMENT_TYPE_DOUBLE) {

  protected def toLong(k : Long) : Long  = k

}
